### PR TITLE
feat: impl for semver::Version

### PR DESCRIPTION
**Pull request by tymmesyde**

### CODE REVIEW

**Comment by escritorio-gustavo** [comment]
Hey, can you add some tests and fix the merge conflicts?

File: ts-rs/tests/semver.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#![allow(deprecated)]
+#![cfg(feature = "chrono-impl")]
```
Comment (escritorio-gustavo) [inline ts-rs/tests/semver.rs:2]: I think this is supposed to be `#![cfg(feature = "semver-impl")]` right?

File: ts-rs/tests/semver.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#![allow(deprecated)]
```
Comment (escritorio-gustavo) [inline ts-rs/tests/semver.rs:1]: Just curious, what is deprecated in this file?

**Review by escritorio-gustavo — COMMENTED** [review]
I am confused about the attributes on your test file

File: ts-rs/tests/semver.rs (hunk 1 above)
Comment (tymmesyde) [reply ts-rs/tests/semver.rs:2]: Yeah my bad

File: ts-rs/tests/semver.rs (hunk 2 above)
Comment (tymmesyde) [reply ts-rs/tests/semver.rs]: I just forgot to remove this line sorry

**Review by escritorio-gustavo — APPROVED** [review]
LGTM, thanks for the PR!

**Pull request merged**